src/main/thrift/iam.thrift

```diff
@@ -0,0 +1,16 @@
+namespace java bio.terra.common.iam.thrift
+
+struct AuthenticatedUserRequestModel {
```

Comment:
One interesting to see (and also pain using Swagger as Pojo) is polymorphism support
I didn't find a clear built-in support in thrift yet, is that simple?

Reply:
By polymorphism do you mean having the ability to build messages off a common base message at the IDL level?  Thrift supports this for RPC services but not serialized data types (`struct`); generally this is approximated using composition.

Or do you mean polymorphism in the generated code (which it does, but it's messy; you can see in the signatures of the `putMessage()`/`getMessage()` methods in `AuthenticatedUserRequest`.